### DISCUSSION TITLE
Openzeppelin path not working

### DISCUSSION THREAD

**Discussion by DukeLurkin** [Cyfrin/foundry-full-course-cu#1346] [Q&A]
I've got the path copied right from the cyfrin repo in HelperConfig but it says "Source "lib/openzeppelin-contracts/contracts/mocks/ERC20Mock.sol" not found: File not found. Searched the following locations: "". "  Maybe openzeppelin changed their file structures? 

<img width="662" alt="helperConfigOpenZepError" src="https://github.com/Cyfrin/foundry-full-course-f23/assets/123668611/03ab7e45-db1a-43ee-bd01-0dc4500946b7">

**Comment by 0xusmanf** [comment]
Yes in the latest version, they have moved it to the following path. 
```solidity
"@openzeppelin-contracts/contracts/mocks/token"
```
Alternately you should use `forge install openzeppelin/openzeppelin-contracts@v4.8.3 --no-commit` command to download the appropriate version synced with the video.

**Reply by tobezhanabi** [reply]
correct

**Comment by DukeLurkin** [comment]
So I did all that and  it says that path is still wrong AND now reentrancyguard is wrong.

new install:

<img width="1005" alt="OpenZepInstallAgain" src="https://github.com/Cyfrin/foundry-full-course-f23/assets/123668611/f2b475d5-b0a6-42f5-b147-2ff08299677c">

same import with bad path:

<img width="594" alt="UpdatedErc20StillWrong" src="https://github.com/Cyfrin/foundry-full-course-f23/assets/123668611/763920b4-c009-46d6-b2a9-3410d69b9027">

Didn't change import but after install it's now bad:

<img width="720" alt="NewRentrancyGuardPathBad" src="https://github.com/Cyfrin/foundry-full-course-f23/assets/123668611/39266921-0c95-4ec0-b162-b06129a0b3eb">

**Reply by DukeLurkin** [reply]
Adding .sol after "token" in ERC20mock didn't work either.

**Reply by tobezhanabi** [reply]
Try forge build first. Sometimes VsCODE doesn't pick up such directories

**Reply by DukeLurkin** [reply]
I've tried a few things and now I'm stuck on a custom error:

<img width="914" alt="setupFailedCustomError92266" src="https://github.com/Cyfrin/foundry-full-course-f23/assets/123668611/549120bd-668f-459d-bdfe-d2b1be214761">

**Reply by tobezhanabi** [reply]
Check your setup function in the test file.

Are you importing Test from forge-std

**Reply by DukeLurkin** [reply]
Yup, up top with all the other imports: import {Test} from "forge-std/Test.sol";

**Reply by 0xusmanf** [reply]
Please run `forge test -vvvvv` and post the log here so we can Identify the root cause of this error, it would be more convenient for us if you can provide the GitHub link to this repo too.

**Reply by DukeLurkin** [reply]
repo:
https://github.com/DukeLurkin/foundry-defi-stablecoin-f23

<img width="974" alt="TestFailGetUsdValue" src="https://github.com/Cyfrin/foundry-full-course-f23/assets/123668611/ce404569-33f2-495d-b532-3dc015aa90fc">


it says unreachable code on the ReentrancyGuard import but that path looks right in my code.

**Reply by 0xusmanf** [reply]
First you have import error at [this line](https://github.com/DukeLurkin/foundry-defi-stablecoin-f23/blob/92e0cb211f172071fb030aae38ef6cd6b2d7139a/src/DSCEngine.sol#L29), make the below changes to resolve.
```diff
- import {ReentrancyGuard} from "@openzeppelin-contracts/contracts/security/ReentrancyGuard.sol";
+ import {ReentrancyGuard} from "@openzeppelin/contracts/security/ReentrancyGuard.sol";
```
Second your are receiving `wbtcUsdPriceFeed` instead of `weth` in your test setup at [this line](https://github.com/DukeLurkin/foundry-defi-stablecoin-f23/blob/92e0cb211f172071fb030aae38ef6cd6b2d7139a/test/unit/DSCEngineTest.t.sol#L22), which is why your test is reverting because `wbtcUsdPriceFeed` returns zero address when inquired from `s_priceFeeds` mapping hence calling latestRoundData from a zero address would revert. Make the below changes to resolve.
```diff
- (ethUsdPriceFee, weth, , , ) = config.activeNetworkConfig();
+ (ethUsdPriceFee, , weth, , ) = config.activeNetworkConfig();
```

**Reply by DukeLurkin** [reply]
Good news everyone!  The test passes but I still have unreachable code in the ReentrancyGuard.  The whole point of that is to stop malicious actors, correct?  I probably should have that working for a real project.  

<img width="892" alt="testPassUnreachableCodeReentrancyGuard" src="https://github.com/Cyfrin/foundry-full-course-f23/assets/123668611/33efb588-775c-47d4-84ce-1f4bd13498ba">


Changed Reentrancy to this but it still won't work:

<img width="767" alt="ReentrancyGuardChangedStillUnreachable" src="https://github.com/Cyfrin/foundry-full-course-f23/assets/123668611/fdc9a0f1-a1f6-4da8-a5c3-2a9a2bce0af0">

**Comment by DukeLurkin** [comment]
For whatever reason this seems to be my paths:
import {ERC20Mock} from "@openzeppelin/contracts/mocks/ERC20Mock.sol";
&
import {ReentrancyGuard} from "@openzeppelin/contracts/security/ReentrancyGuard.sol";

Hopefully this doesn't screw something else up down the line, idk?

**Comment by DukeLurkin** [comment]
Now I'm here with this problem, pretty sure it's the same issue:

<img width="777" alt="Unreachable" src="https://github.com/Cyfrin/foundry-full-course-f23/assets/123668611/2791e9f0-c8a9-43b6-bba6-cff5073a8c69">
